[Bug]: 9.0.0-beta.8 not type compatible with React 19 (react-router-dom)

### Describe the bug

Something has changed between beta.7 and beta.8 that means the storybook router bundle is now being imported - and it contains types incompatible with react 19.

```
.yarn/__virtual__/storybook-virtual-2ecb765cb8/3/AppData/Local/Yarn/Berry/cache/storybook-npm-9.0.0-beta.8-4eb86a612e-10c0.zip/node_modules/storybook/dist/router/index.d.ts:303:85 - error TS2503: Cannot find namespace 'JSX'.

303 declare function BrowserRouter({ basename, children, window }: BrowserRouterProps): JSX.Element;
                                                                                        ~~~


Found 1 error in .yarn/__virtual__/storybook-virtual-2ecb765cb8/3/AppData/Local/Yarn/Berry/cache/storybook-npm-9.0.0-beta.8-4eb86a612e-10c0.zip/node_modules/storybook/dist/router/index.d.ts:303
```

![Image](https://github.com/user-attachments/assets/c3ef7974-d7bc-461c-bc78-9e4cdf92d209)

### Reproduction link

-

### Reproduction steps

_No response_

### System

```bash
Storybook Environment Info:

  System:
    OS: Windows 10 10.0.19045
    CPU: (24) x64 Intel(R) Core(TM) i9-10920X CPU @ 3.50GHz
  Binaries:
    Node: 22.15.0 - C:\Program Files\nodejs\node.EXE
    Yarn: 4.9.1 - C:\Program Files\nodejs\yarn.CMD <----- active
    npm: 10.9.2 - C:\Program Files\nodejs\npm.CMD
  Browsers:
    Edge: Chromium (135.0.3179.98)
```

### Additional context

_No response_

--- a/code/core/package.json
+++ b/code/core/package.json
@@ -746,7 +746,7 @@
     "react-helmet-async": "^1.3.0",
     "react-inspector": "^6.0.0",
     "react-popper-tooltip": "^4.4.2",
-    "react-router-dom": "6.0.2",
+    "react-router-dom": "6.15.0",
     "react-syntax-highlighter": "^15.4.5",
     "react-textarea-autosize": "^8.3.0",
     "react-transition-group": "^4.4.5",

